### PR TITLE
fix API incompatibility introduced in 3.1.0

### LINKED ISSUE #4443: realm getSchema incorrect return type
realm.getSchema() function return object from type StandardRealmSchema 
this type is object have packcge only access so it is impossible to access his methods,
the only way is to convert this object to (RealmSchema), i think it is mistake 
the return type need to be RealmSchema and not StandardRealmSchema 

realm version in use :3.1.0

### PR DESCRIPTION
**Pull request by zaki50**
fixes #4443 

I fixed incompatibility of `RealmSchema.getAll()` in addition to `BaseRealm.getSchema()`.

@realm/java

### CODE REVIEW

File: realm/realm-library/src/main/java/io/realm/OsRealmSchema.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
  */
```
Comment (bmeike) [inline realm/realm-library/src/main/java/io/realm/OsRealmSchema.java]: I don't think you should change this class.  `OsRealmSchema` can not handle generic `RealmObjectSchema`.  It is only able to handle `OsRealmObjectSchema`.  It should use typing to make that clear.

File: realm/realm-library/src/main/java/io/realm/RealmSchema.java [hunk 2]
```diff
@@ -54,7 +54,7 @@
      *
      * @return the set of all classes in this Realm or no RealmObject classes can be saved in the Realm.
      */
-    public abstract Set<? extends RealmObjectSchema> getAll();
```
Comment (bmeike) [inline realm/realm-library/src/main/java/io/realm/RealmSchema.java]: I believe this is correct as it stands.

**Comment by bmeike** [comment]
Thanks for fixing this.

**Comment by bmeike** [comment]
Hey @zaki50, let me know if you agree with my suggestion that we ought to keep the more specific type in OsRealmSchema.  Also, happy to discuss if you want.  A million thanks for fixing these bugs I introduced.

File: realm/realm-library/src/main/java/io/realm/RealmSchema.java (hunk 2 above)
Comment (zaki50) [inline realm/realm-library/src/main/java/io/realm/RealmSchema.java]: The point is that we can't introduce any breaking changes without bumping major version.

Changing return type from `Set <RealmObjectSchema>` to `Set<? extends RealmObjectSchema>` is a breaking change. Users might have a code `Set<RealmObjectSchema> allObjectSchema = realm.getSchema().getAll();` and this line causes compile error in 3.1.

Comment (bmeike) [inline realm/realm-library/src/main/java/io/realm/RealmSchema.java]: Yep, I guess that is right.  That is a serious bummer.

**Pull request merged**